openghg/store/base/_base.py

```diff
@@ -220,19 +244,16 @@ def get_lookup_keys(self, optional_metadata: Optional[Dict]) -> List[str]:
         optional = metakeys.get("optional", {})
 
         lookup_keys = list(required)
-        # Check if anything in optional_metadata tries to override our required keys
-        if optional_metadata is not None:
-            common_keys = set(required) & set(optional_metadata)
 
-            if common_keys:
-                raise ValueError(
-                    f"The following optional metadata keys are already present in required keys: {', '.join(common_keys)}"
-                )
+        # Note: Just grabbing the first entry in data at the moment
```

Comment:
Now that this doesn't depend on passing `optional_metadata`, it could be incorporated into `datasource_lookup`, so it could be call on each entry separately. I think it could be called around line 380 of `_base.py`. (So `required_keys` wouldn't be passed to `assign_data`.)

Reply:
Good idea - I'd updated `assign_data` and the data type classes to accommodate this.

I did leave `required_keys` input but it is now optional and uses the output of `get_lookup_keys` by default.

Reply:
I guess this would be a bigger change, but the datasource lookup should probably be done on each item in `data`. Maybe I'll open an issue for that. Then we know that the required keys are specific to the data for one ("proto") datasource.

Reply:
Yes I know what you mean - before we were always assuming the same *keys* would be used for lookup with different values, and I believe in most cases that would still be true, but we could have the occasional situation where its not true. I don't believe we have any examples of that yet though. New issue for this makes sense to keep track but perhaps low priority for now.